src/AVFoundation/AVCompat.cs

```diff
@@ -169,6 +169,24 @@ public AVAudioUnitComponentManager ()
 		}
 	}
 
+	public delegate int AVAudioSourceNodeRenderHandler (bool isSilence, AudioToolbox.AudioTimeStamp timestamp, uint frameCount, ref AudioToolbox.AudioBuffers outputData);
+
+	partial class AVAudioSourceNode {
+		[Obsolete("Use 'AVAudioSourceNode (AVAudioSourceNodeRenderHandler2)' instead.")]
+		public AVAudioSourceNode (AVAudioSourceNodeRenderHandler renderHandler)
+			: base (NSObjectFlag.Empty)
+		{
+			throw new NotImplementedException ();
```

Comment:
I think we can show a better exception here:

```suggestion
			throw new InvalidOperationException ("Do not use this constructor. Use the 'AVAudioSourceNode (AVAudioSourceNodeRenderHandler2)' constructor instead.");
```